tfhe/src/integer/server_key/radix_parallel/div_mod.rs

```diff
@@ -0,0 +1,496 @@
+use crate::integer::ciphertext::RadixCiphertext;
+use crate::integer::server_key::comparator::ZeroComparisonType;
+use crate::integer::ServerKey;
+use rayon::prelude::*;
+
+use super::bit_extractor::BitExtractor;
+
+impl ServerKey {
+    //======================================================================
+    //                Div Rem
+    //======================================================================
+    pub fn unchecked_div_rem_parallelized(
+        &self,
+        numerator: &RadixCiphertext,
+        divisor: &RadixCiphertext,
+    ) -> (RadixCiphertext, RadixCiphertext) {
+        // Pseudo-code of the algorithm:
+        //
+        //
+        // div(N/D):
+        // Q := 0                  -- Initialize quotient and remainder to zero
+        // R := 0
+        // for i := n − 1 .. 0 do  -- Where n is number of bits in N
+        //   R := R << 1           -- Left-shift R by 1 bit
+        //   R(0) := N(i)          -- Set the least-significant bit of R equal to bit i of the
+        //                         -- numerator
+        //   if R ≥ D then
+        //     R := R − D
+        //     Q(i) := 1
+        //   end
+        // end
+        let num_blocks = numerator.blocks.len();
+        let num_bits_in_block = (self.key.message_modulus.0 as f64).log2() as u64;
+        let total_bits = num_bits_in_block * num_blocks as u64;
+
+        let mut quotient = self.create_trivial_zero_radix(num_blocks);
+        let mut remainder = self.create_trivial_zero_radix(num_blocks);
+
+        // This lut only works when y values are 0 or 1
+        let zeroer_lut_for_merged_cmp =
```

Comment:
zeroer ?

Reply:
This lut is used to zero out or not a ciphertext depending on a condition

Reply:
thx